src/modules/exchange/echange.abstract.ts

```diff
@@ -0,0 +1,87 @@
+import axios from 'axios';
+import { HttpException, HttpStatus } from '@nestjs/common';
+
+import { RateResponce } from './dto/exchange.dto';
+
+export enum Provider {
+  ExchangeRate = 'ExchangeRate',
+  PrivatBank = 'PrivatBank',
+  MonoBank = 'MonoBank',
+}
+
+export interface ExchangeRateData {
+  conversion_rates: {
+    USD: number;
+  };
+}
+
+export interface PrivatBankData {
+  ccy: string;
+  base_ccy: string;
+  buy: string;
+  sale: string;
+}
+
+export interface MonoBankData {
+  currencyCodeA: number;
+  currencyCodeB: number;
+  rateBuy: number;
+  rateSell: number;
+}
+
+type CurrencyData = ExchangeRateData | PrivatBankData[] | MonoBankData[];
```

Comment:
Shouldn't abstract class include only abstract entities and not details? these interfaces belong to their own classes
Now every time you add another provider - you would go here to add another interface

Reply:
I agree with you on this, but I find lesser evil then return "any"

Reply:
But you could at least export the type from a separate place, that would eliminate the need for this file editing on adding a new provider